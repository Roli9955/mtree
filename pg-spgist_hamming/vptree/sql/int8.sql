-- int8 check

-- SELECT pg_sleep(20);

CREATE TABLE int8tmp_2 (a int8);

\copy int8tmp_2 from 'data/int8.data'
\copy int8tmp_2 from 'data/incr.data'

SET enable_seqscan=on;

SELECT count(*) FROM int8tmp_2 WHERE a <  464571291354841;

SELECT count(*) FROM int8tmp_2 WHERE a <= 464571291354841;

SELECT count(*) FROM int8tmp_2 WHERE a  = 464571291354841;

SELECT count(*) FROM int8tmp_2 WHERE a >= 464571291354841;

SELECT count(*) FROM int8tmp_2 WHERE a >  464571291354841;

-- These succeed, since the system falls back
-- to a sequential scan because of the lack of support
-- for these operators in the index
SELECT count(*) FROM int8tmp WHERE a  < 7::int8;
SELECT count(*) FROM int8tmp WHERE a  > 7::int8;
SELECT count(*) FROM int8tmp WHERE a  <= 7::int8;
SELECT count(*) FROM int8tmp WHERE a  >= 7::int8;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM int8tmp WHERE a  < 7::int8;

SELECT a, a <@ ('5', 1), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 1);
SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 2);
SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 3);
SELECT a, a <@ ('5', 4), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 4);

SELECT a, a <@ (5, 1), a <-> 5 FROM int8tmp WHERE  a <@ (5, 1);
SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 2);
SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 3);
SELECT a, a <@ (5, 4), a <-> 5 FROM int8tmp WHERE  a <@ (5, 4);


CREATE INDEX int8idx_2 ON int8tmp_2 USING spgist ( a vptree_ops );

SET enable_seqscan=off;

-- These should fail, since we don't provide operators
-- for the relevant conditionals (<, >, <=, >=),
-- and sequential-scans are disabled.
-- Apparently the SP-GiST system just coerces the
-- result to an empty set if it doesn't know how
-- to apply the operator
SELECT count(*) FROM int8tmp WHERE a  < 7::int8;
SELECT count(*) FROM int8tmp WHERE a  > 7::int8;
SELECT count(*) FROM int8tmp WHERE a  <= 7::int8;
SELECT count(*) FROM int8tmp WHERE a  >= 7::int8;

SELECT a, a <@ ('5', 1), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 1);
SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 2);
SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 3);
SELECT a, a <@ ('5', 4), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 4);

SELECT a, a <@ (5, 1), a <-> 5 FROM int8tmp WHERE  a <@ (5, 1);
SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 2);
SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 3);
SELECT a, a <@ (5, 4), a <-> 5 FROM int8tmp WHERE  a <@ (5, 4);

SELECT a, a <@ (93382613193471632, 1000000), a <-> 93382613193471632 FROM int8tmp_2 WHERE a <@ (93382613193471632, 1000000) ORDER BY a <-> 93382613193471632 LIMIT 3;

SELECT a, a <@ (5, 4), a <-> 5 FROM int8tmp_2 WHERE a <@ (5, 4) ORDER BY a <-> 5 LIMIT 3;

EXPLAIN (COSTS OFF)
SELECT a, a <@ (64571291354841, 100), a <-> 64571291354841 FROM int8tmp_2 WHERE a <@ (64571291354841, 100) ORDER BY a <-> 64571291354841 LIMIT 3;

EXPLAIN (COSTS OFF)
SELECT a, a <@ (464571291354841, 100) FROM int8tmp_2 WHERE a <@ (464571291354841, 100);
