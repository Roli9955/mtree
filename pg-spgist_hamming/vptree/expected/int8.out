-- int8 check
-- SELECT pg_sleep(20);
CREATE TABLE int8tmp_2 (a int8);
\copy int8tmp_2 from 'data/int8.data'
\copy int8tmp_2 from 'data/incr.data'
SET enable_seqscan=on;
SELECT count(*) FROM int8tmp_2 WHERE a <  464571291354841;
 count 
-------
   776
(1 row)

SELECT count(*) FROM int8tmp_2 WHERE a <= 464571291354841;
 count 
-------
   777
(1 row)

SELECT count(*) FROM int8tmp_2 WHERE a  = 464571291354841;
 count 
-------
     1
(1 row)

SELECT count(*) FROM int8tmp_2 WHERE a >= 464571291354841;
 count 
-------
   271
(1 row)

SELECT count(*) FROM int8tmp_2 WHERE a >  464571291354841;
 count 
-------
   270
(1 row)

-- These succeed, since the system falls back
-- to a sequential scan because of the lack of support
-- for these operators in the index
SELECT count(*) FROM int8tmp WHERE a  < 7::int8;
 count 
-------
     7
(1 row)

SELECT count(*) FROM int8tmp WHERE a  > 7::int8;
 count 
-------
     2
(1 row)

SELECT count(*) FROM int8tmp WHERE a  <= 7::int8;
 count 
-------
     8
(1 row)

SELECT count(*) FROM int8tmp WHERE a  >= 7::int8;
 count 
-------
     3
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM int8tmp WHERE a  < 7::int8;
            QUERY PLAN             
-----------------------------------
 Aggregate
   ->  Seq Scan on int8tmp
         Filter: (a < '7'::bigint)
(3 rows)

SELECT a, a <@ ('5', 1), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 1);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 4 | t        |        1
(3 rows)

SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 2);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 7 | t        |        2
 3 | t        |        2
 4 | t        |        1
(5 rows)

SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 3);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 2 | f        |        3
 7 | t        |        2
 8 | f        |        3
 3 | t        |        2
 4 | t        |        1
(7 rows)

SELECT a, a <@ ('5', 4), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 4);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 1 | t        |        4
 2 | t        |        3
 7 | t        |        2
 8 | t        |        3
 3 | t        |        2
 4 | t        |        1
 9 | t        |        4
(9 rows)

SELECT a, a <@ (5, 1), a <-> 5 FROM int8tmp WHERE  a <@ (5, 1);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 4 | t        |        1
(3 rows)

SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 2);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 7 | t        |        2
 3 | t        |        2
 4 | t        |        1
(5 rows)

SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 3);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 2 | f        |        3
 7 | t        |        2
 8 | f        |        3
 3 | t        |        2
 4 | t        |        1
(7 rows)

SELECT a, a <@ (5, 4), a <-> 5 FROM int8tmp WHERE  a <@ (5, 4);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 1 | t        |        4
 2 | t        |        3
 7 | t        |        2
 8 | t        |        3
 3 | t        |        2
 4 | t        |        1
 9 | t        |        4
(9 rows)

CREATE INDEX int8idx_2 ON int8tmp_2 USING spgist ( a vptree_ops );
SET enable_seqscan=off;
-- These should fail, since we don't provide operators
-- for the relevant conditionals (<, >, <=, >=),
-- and sequential-scans are disabled.
-- Apparently the SP-GiST system just coerces the
-- result to an empty set if it doesn't know how
-- to apply the operator
SELECT count(*) FROM int8tmp WHERE a  < 7::int8;
 count 
-------
     0
(1 row)

SELECT count(*) FROM int8tmp WHERE a  > 7::int8;
 count 
-------
     0
(1 row)

SELECT count(*) FROM int8tmp WHERE a  <= 7::int8;
 count 
-------
     0
(1 row)

SELECT count(*) FROM int8tmp WHERE a  >= 7::int8;
 count 
-------
     0
(1 row)

SELECT a, a <@ ('5', 1), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 1);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 4 | t        |        1
(3 rows)

SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 2);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 7 | t        |        2
 3 | t        |        2
 4 | t        |        1
(5 rows)

SELECT a, a <@ ('5', 2), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 3);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 2 | f        |        3
 7 | t        |        2
 8 | f        |        3
 3 | t        |        2
 4 | t        |        1
(7 rows)

SELECT a, a <@ ('5', 4), a <-> 5 FROM int8tmp WHERE  a <@ ('5', 4);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 1 | t        |        4
 2 | t        |        3
 7 | t        |        2
 8 | t        |        3
 3 | t        |        2
 4 | t        |        1
 9 | t        |        4
(9 rows)

SELECT a, a <@ (5, 1), a <-> 5 FROM int8tmp WHERE  a <@ (5, 1);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 4 | t        |        1
(3 rows)

SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 2);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 7 | t        |        2
 3 | t        |        2
 4 | t        |        1
(5 rows)

SELECT a, a <@ (5, 2), a <-> 5 FROM int8tmp WHERE  a <@ (5, 3);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 2 | f        |        3
 7 | t        |        2
 8 | f        |        3
 3 | t        |        2
 4 | t        |        1
(7 rows)

SELECT a, a <@ (5, 4), a <-> 5 FROM int8tmp WHERE  a <@ (5, 4);
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 6 | t        |        1
 1 | t        |        4
 2 | t        |        3
 7 | t        |        2
 8 | t        |        3
 3 | t        |        2
 4 | t        |        1
 9 | t        |        4
(9 rows)

SELECT a, a <@ (93382613193471632, 1000000), a <-> 93382613193471632 FROM int8tmp_2 WHERE a <@ (93382613193471632, 1000000) ORDER BY a <-> 93382613193471632 LIMIT 3;
         a         | ?column? | ?column? 
-------------------+----------+----------
 93382613193471632 | t        |        0
(1 row)

SELECT a, a <@ (5, 4), a <-> 5 FROM int8tmp_2 WHERE a <@ (5, 4) ORDER BY a <-> 5 LIMIT 3;
 a | ?column? | ?column? 
---+----------+----------
 5 | t        |        0
 4 | t        |        1
 6 | t        |        1
(3 rows)

EXPLAIN (COSTS OFF)
SELECT a, a <@ (64571291354841, 100), a <-> 64571291354841 FROM int8tmp_2 WHERE a <@ (64571291354841, 100) ORDER BY a <-> 64571291354841 LIMIT 3;
                                       QUERY PLAN                                        
-----------------------------------------------------------------------------------------
 Limit
   ->  Sort
         Sort Key: ((a <-> '64571291354841'::bigint))
         ->  Index Only Scan using int8idx_2 on int8tmp_2
               Index Cond: (a <@ ROW('64571291354841'::bigint, '100'::double precision))
(5 rows)

EXPLAIN (COSTS OFF)
SELECT a, a <@ (464571291354841, 100) FROM int8tmp_2 WHERE a <@ (464571291354841, 100);
                                  QUERY PLAN                                  
------------------------------------------------------------------------------
 Index Only Scan using int8idx_2 on int8tmp_2
   Index Cond: (a <@ ROW('464571291354841'::bigint, '100'::double precision))
(2 rows)

